### PR TITLE
Kubernetes

### PR DESCRIPTION
**Pull request by ShahNewazKhan**
- Added support to get aws creds from prompt if `config.kubernetes.deploy = true`
- Resolved x509 cacert issue for docker

https://github.com/commitdev/commit0/issues/66

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -36,19 +36,27 @@ RUN mkdir -p /tmp/protoc && \
 
 RUN GO111MODULE=off go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
 
+RUN curl -kLo /tmp/terraform.zip "https://releases.hashicorp.com/terraform/0.12.12/terraform_0.12.12_linux_amd64.zip" && \
```
Comment (bmonkman) [inline Dockerfile]: Please remove -k and add -sS

File: internal/generate/kubernetes/generate.go [hunk 2]
```diff
@@ -1,43 +1,84 @@
...
 	cmd.Dir = fmt.Sprintf("%s/kubernetes/terraform/environments/staging", dir)
 
 	stdoutPipe, _ := cmd.StdoutPipe()
 	stderrPipe, _ := cmd.StderrPipe()
 
 	var errStdout, errStderr error
-	err = cmd.Start()
+
+	if envars != nil {
+		log.Println("Setting envars to cmd ...")
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: I don't think this is worth logging

File: internal/generate/kubernetes/generate.go [hunk 3]
```diff
@@ -62,3 +103,141 @@ func execute(cmd *exec.Cmd) {
...
+		log.Fatalln(err)
+	}
+
+	return awsSecrets
+}
+
+func writeSecrets(s Secrets) {
+	secretsYaml, err := yaml.Marshal(&s)
+
+	fmt.Printf("SECRETS: %v", string(secretsYaml))
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: Was this just for debugging?

File: internal/generate/kubernetes/generate.go [hunk 4]
```diff
@@ -62,3 +103,141 @@ func execute(cmd *exec.Cmd) {
...
+
+	if err != nil {
+		log.Fatalf("error: %v", err)
+		panic(err)
+	}
+
+	dir, err := os.Getwd()
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: Not using your helper func

File: internal/generate/kubernetes/generate.go [hunk 5]
```diff
@@ -1,43 +1,84 @@
...
+	}
+}
+
+// Generate k8s templates
 func Generate(t *templator.Templator, cfg *config.Commit0Config, wg *sync.WaitGroup) {
 	data := templator.GenericTemplateData{*cfg}
 	t.Kubernetes.TemplateFiles(data, false, wg)
 }
 
+func getCwd() string {
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: Maybe move this to util?

File: internal/generate/kubernetes/generate.go [hunk 6]
```diff
@@ -62,3 +103,141 @@ func execute(cmd *exec.Cmd) {
...
+	}
+
+	dir, err := os.Getwd()
+
+	if err != nil {
+		log.Fatalf("Getting working directory failed: %v\n", err)
+		panic(err)
+	}
+
+	secretsFile := fmt.Sprintf("%s/secrets.yaml", dir)
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: You can use `fikepath.Join` to create paths in a portable way

File: internal/generate/kubernetes/generate.go [hunk 7]
```diff
@@ -62,3 +103,141 @@ func execute(cmd *exec.Cmd) {
...
+	// err = ioutil.WriteFile(secretsFile, []byte(secretsYaml), 0644)
+	f, err := os.Create(secretsFile)
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: Why not just use WriteFile?

File: internal/generate/kubernetes/generate.go [hunk 8]
```diff
@@ -62,3 +103,141 @@ func execute(cmd *exec.Cmd) {
...
+	if err != nil {
+		log.Fatalf("error: %v", err)
+		panic(err)
+	}
+
+	defer f.Close()
+
+	n3, err := f.WriteString(string(secretsYaml))
+	f.Sync()
+	log.Printf("Wrote %d bytes to %v", n3, secretsFile)
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: Too much info

File: internal/generate/kubernetes/generate.go [hunk 9]
```diff
@@ -62,3 +103,141 @@ func execute(cmd *exec.Cmd) {
...
+		var awsAccessKeyIDPat = regexp.MustCompile(`^[A-Z0-9]{20}$`)
+		if !awsAccessKeyIDPat.MatchString(input) {
+			return errors.New("Invalid aws_access_key_id")
+		}
+		return nil
+	}
+
+	validateSAK := func(input string) error {
+		// 40 base64 characters
+		var awsAccessKeyIDPat = regexp.MustCompile(`^[A-Za-z0-9/+=]{40}$`)
```
Comment (bmonkman) [inline internal/generate/kubernetes/generate.go]: Wrong var name

File: go.sum [hunk 10]
```diff
@@ -1,5 +1,18 @@
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/alecthomas/gometalinter v2.0.11+incompatible h1:ENdXMllZNSVDTJUUVIzBW9CSEpntTrQa76iRsEFLX/M=
```
Comment (bmonkman) [inline go.sum]: Did all these new deps come from the prompt lib?

File: internal/generate/kubernetes/generate.go (hunk 3 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: Correct, removed

File: internal/generate/kubernetes/generate.go (hunk 2 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: Removed

File: Dockerfile (hunk 1 above)
Comment (ShahNewazKhan) [reply Dockerfile]: Good catch! Updated accordingly

File: internal/generate/kubernetes/generate.go (hunk 4 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: Updated

File: internal/generate/kubernetes/generate.go (hunk 5 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: Moved to util

File: go.sum (hunk 10 above)
Comment (ShahNewazKhan) [reply go.sum]: Correct https://github.com/manifoldco/promptui/blob/master/go.mod#L4

File: internal/generate/kubernetes/generate.go (hunk 8 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: Removed

File: internal/generate/kubernetes/generate.go (hunk 7 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: Reverted to WriteFile

File: internal/generate/kubernetes/generate.go (hunk 6 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: Changed to `filepath.Join`

File: internal/generate/kubernetes/generate.go (hunk 9 above)
Comment (ShahNewazKhan) [reply internal/generate/kubernetes/generate.go]: changed to awsSecretAccessKeyPat

**Comment by ShahNewazKhan** [comment]
@bmonkman updated as per review comments

**Pull request merged**